apps/condo/domains/user/oidc/adapter/GQLAdapter.js

```diff
@@ -0,0 +1,260 @@
+         * - params {object} - parsed recognized parameters object
+         * - lastSubmission {object} - previous interaction result submission
+         * - trusted {string[]} - parameter names that come from a trusted source
+         * - result {object} - interaction results object is expected here
+         * - grantId {string} - grant identifier if there's a preexisting one
+         * - session {object}
+         * - session.uid {string} - uid of the session this Interaction belongs to
+         * - session.cookie {string} - jti of the session this Interaction belongs to
+         * - session.acr {string} - existing acr of the session Interaction belongs to
+         * - session.amr {string[]} - existing amr of the session Interaction belongs to
+         * - session.accountId {string} - existing account id from the seession Interaction belongs to
+         *
+         * Replay prevention ReplayDetection model contains the following properties:
+         * - jti {string} - unique identifier of the replay object
+         * - kind {string} - "ReplayDetection" fixed string value
+         * - exp {number} - timestamp of the replay object cache expiration
+         * - iat {number} - timestamp of the replay object cache's creation
+         */
+        const expiresAt = expiresIn ? new Date(Date.now() + expiresIn * 1000) : undefined
+        const dvAndSender = {
+            dv: 1,
+            sender: {
+                dv: 1,
+                fingerprint: 'create-oidc-client',
+            },
+        }
+        await OidcClient.create(this.context, { ...dvAndSender, clientId: id, payload, expiresAt })
+    }
+
+    /**
+     *
+     * Return previously stored instance of an oidc-provider model.
+     *
+     * @return {Promise} Promise fulfilled with what was previously stored for the id (when found and
+     * not dropped yet due to expiration) or falsy value when not found anymore. Rejected with error
+     * when encountered.
+     * @param {string} id Identifier of oidc-provider model
+     *
+     */
+    async find (id) {
+        const item = await OidcClient.getOne(this.context, { clientId: id })
+        if (!item) {
+            throw new Error(`There is no OIDC client with clientId=${id}`)
+        }
+        return item.payload
+    }
+
+    /**
+     *
+     * Return previously stored instance of DeviceCode by the end-user entered user code. You only
+     * need this method for the deviceFlow feature
+     *
+     * @return {Promise} Promise fulfilled with the stored device code object (when found and not
+     * dropped yet due to expiration) or falsy value when not found anymore. Rejected with error
+     * when encountered.
+     * @param {string} userCode the user_code value associated with a DeviceCode instance
+     *
+     */
+    async findByUserCode (userCode) {
+        return Promise.reject(new Error('Method is not needed in this adapter'))
```

Comment:
Should this method be abstract? Or maybe it shouldn't be defined at all.

Reply:
If we erroneously start using GQLAdapter to find something different from 'Client' we will get a particular error. That's the reason why I create some stubs for methods.